### PR TITLE
k6: 0.38.3 -> 0.39.0

### PR DESCRIPTION
**Pull request by bryanasdev000**
###### Description of changes

- bump version
- drop explict docheck
- add shell completion for bash,fish and zsh

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/k6/default.nix [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
 
-  doCheck = true;
+  nativeBuildInputs = [ installShellFiles ];
+
   doInstallCheck = true;
   installCheckPhase = ''
     $out/bin/k6 version | grep ${version} > /dev/null
   '';
 
+  postInstall = ''
```
Comment (Mic92) [inline pkgs/development/tools/k6/default.nix:25]: To not break cross compilation:

```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
```

Comment (bryanasdev000) [reply pkgs/development/tools/k6/default.nix:25]: Welp, can you explain/reference a bit this part?

I have a postInstall to install shell files in a bunch of other Golang based projects.

Comment (smancill) [reply pkgs/development/tools/k6/default.nix:25]: I guess it is because you are using `$out/bin/k6` in the hook.

Comment (Mic92) [reply pkgs/development/tools/k6/default.nix:25]: Yes. If you cross compile i.e. hostPlatform is aarch64 and buildPlatform is x86_64, than you cannot run your $out/bin/k6 binary compiled for aarch64 on x86_64.

Comment (bryanasdev000) [reply pkgs/development/tools/k6/default.nix]: Thanks @smancill @Mic92, will keep in mind to refactor my packages.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 182772` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>k6</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 182772` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>k6</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**